src/MySqlConnector/Serialization/ColumnDefinitionPayload.cs

```diff
@@ -35,17 +51,90 @@ public static ColumnDefinitionPayload Create(PayloadData payload)
 			}
 
 			if (reader.BytesRemaining != 0)
+			{
 				throw new FormatException("Extra bytes at end of payload.");
-			return new ColumnDefinitionPayload(name, characterSet, columnLength, columnType, columnFlags);
+			}
+
+			return new ColumnDefinitionPayload
+			{
+				Name = name,
+				CharacterSet = characterSet,
+				ColumnLength = columnLength,
+				ColumnType = columnType,
+				ColumnFlags = columnFlags,
+				SchemaName = schema,
+				CatelogName = catalog,
+				Table = table,
+				PhysicalTable = physicalTable,
+				PhysicalName = physicalName,
+				Decimals = decimals
+			};
+
 		}
 
-		private ColumnDefinitionPayload(string name, CharacterSet characterSet, int columnLength, ColumnType columnType, ColumnFlags columnFlags)
+		public Type GetDataType(
+			bool treadTinyAsBoolean = false,
+			bool oldGuids = false)
 		{
-			Name = name;
-			CharacterSet = characterSet;
-			ColumnLength = columnLength;
-			ColumnType = columnType;
-			ColumnFlags = columnFlags;
+			var isUnsigned = (ColumnFlags & ColumnFlags.Unsigned) != 0;
+			switch (ColumnType)
+			{
+				case ColumnType.Tiny:
+					if (treadTinyAsBoolean && ColumnLength == 1)
+					{
+						return typeof(bool);
+					}
+
+					return isUnsigned ? typeof(byte) : typeof(sbyte);
+				case ColumnType.Short:
+					return isUnsigned ? typeof(ushort) : typeof(short);
+				case ColumnType.Int24:
+				case ColumnType.Long:
+					return isUnsigned ? typeof(uint) : typeof(int);
+				case ColumnType.Longlong:
+					return isUnsigned ? typeof(ulong) : typeof(ulong);
+				case ColumnType.Bit:
+					return typeof(ulong);
+				case ColumnType.String:
+					if (!oldGuids && ColumnLength / SerializationUtility.GetBytesPerCharacter(CharacterSet) == 36)
+					{
+						return typeof(Guid);
+					}
+
+					goto case ColumnType.LongBlob;
+
+				case ColumnType.VarString:
+				case ColumnType.VarChar:
+				case ColumnType.TinyBlob:
+				case ColumnType.Blob:
+				case ColumnType.MediumBlob:
+				case ColumnType.LongBlob:
+					if (CharacterSet == CharacterSet.Binary && oldGuids && ColumnLength == 16)
+					{
+						return typeof(Guid);
+					}
+
+					return typeof(string);
```

Comment:
Seems like this should return `typeof(byte[])` when `CharacterSet == CharacterSet.Binary`? Please increase test coverage.

Reply:
Fixed by use `ResultSet.GetFieldType`